js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetNodeMenu.tsx

```diff
@@ -45,27 +46,35 @@ export const useAssetNodeMenu = ({
     onChangeExplorerPath({...explorerPath, opsQuery: nextOpsQuery}, 'push');
   }
 
+  const {
+    featureContext: {canSeeMaterializeAction},
+  } = React.useContext(CloudOSSContext);
+
   return {
     menu: (
       <Menu>
-        <MenuItem
-          icon="materialization"
-          text={
-            <Box flex={{direction: 'row', alignItems: 'center', gap: 4}}>
-              <span>Materialize</span>
-              {loading ? <Spinner purpose="body-text" /> : null}
-            </Box>
-          }
-          onClick={async (e) => {
-            await showSharedToaster({
-              intent: 'primary',
-              message: 'Initiating materialization',
-              icon: 'materialization',
-            });
-            onClick([node.assetKey], e, false);
-          }}
-        />
-        {upstream.length || downstream.length ? <MenuDivider /> : null}
+        {canSeeMaterializeAction ? (
```

Comment:
Ahh I made some related changes in https://github.com/dagster-io/dagster/pull/20982 because this callsite doesn't account for `definition.hasMaterializePermission` - want to merge this and i'll rebase my PR?

Reply:
merged